### PR TITLE
Add completion support for resource access action context

### LINKED ISSUE #36382: Add completion support for resource method call action
**Description:**
$title

https://github.com/ballerina-platform/ballerina-lang/pull/36378 introduce the parser changes for resource method call action. With that the  `remote_action_config3.json` test is disabled since the node is not implemented.

### PR DESCRIPTION
**Pull request by malinthar**
## Purpose
$subject

Fixes #36382 

## Samples
image

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: language-server/modules/langserver-core/src/test/resources/completion/action_node_context/config/client_resource_access_action_config1.json [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+      "detail": "()",
+      "documentation": {
+        "right": {
+          "kind": "markdown",
+          "value": "**Package:** _._  \n  \n  \n**Params**  \n- `string` id1  \n- `string[]` ids  \n- `string` b  \n- `string[]` ids2"
+        }
+      },
+      "sortText": "CC",
+      "filterText": "path1|path2|post",
+      "insertText": "/path1/[${2:\"\"}]/path2/[${4:\"\"}].post(${5})",
```
[the remark this answers is not in the document]
Comment (malinthar) [reply language-server/modules/langserver-core/src/test/resources/completion/action_node_context/config/client_resource_access_action_config1.json:114]: Will do

**Review by mohanvive — DISMISSED** [review]
Some files don't have license header. Please check them and fix.

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/AbstractCompletionProvider.java [hunk 2]
```diff
@@ -181,8 +182,11 @@ protected List<LSCompletionItem> getCompletionItemList(List<? extends Symbol> sc
             if (processedSymbols.contains(symbol)) {
                 return;
             }
-            if (symbol.kind() == FUNCTION || symbol.kind() == METHOD) {
-                completionItems.addAll(populateBallerinaFunctionCompletionItems(symbol, ctx));
+            if (symbol.kind() == FUNCTION || symbol.kind() == METHOD || symbol.kind() == RESOURCE_METHOD) {
+                FunctionSymbol functionSymbol = (FunctionSymbol) symbol;
+                if (functionSymbol.getName().isPresent() && !functionSymbol.getName().get().contains("$")) {
```
[the remark this answers is not in the document]
Comment (IMS94) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/AbstractCompletionProvider.java:187]: Great!

File: language-server/modules/langserver-core/src/test/resources/completion/action_node_context/config/client_resource_access_action_config1.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (malinthar) [reply language-server/modules/langserver-core/src/test/resources/completion/action_node_context/config/client_resource_access_action_config1.json]: Fixed

**Review by IMS94 — APPROVED** [review]

**Review by IMS94 — APPROVED** [review]

**Pull request merged**